From 56fe3d154197ed0e8b8ecdc1f0d59cba2fe1ba16 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 3 Jan 2018 13:51:59 +0800
Subject: [PATCH 3152/5242] MLK-17332: ARM: imx: pm-rpmsg: remove workqueue
 delay

commit  17f3de0f17b678f040edbbade7e8677479355aae from
https://source.codeaurora.org/external/imx/linux-imx.git

With the latest M4 image on i.mx7ULP, which assume life cycle rpmsg is the
first channel sending message during AP bootup, we should remove the delay
timing window which other rpmsg channel may fall in, otherwise, such rpmsg
channel may probe failed as pf1550 regulator rpmsg driver.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Tested-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/pm-rpmsg.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/pm-rpmsg.c b/arch/arm/mach-imx/pm-rpmsg.c
index b4bfa2a..d2a64df 100644
--- a/arch/arm/mach-imx/pm-rpmsg.c
+++ b/arch/arm/mach-imx/pm-rpmsg.c
@@ -230,8 +230,7 @@ static int pm_rpmsg_probe(struct rpmsg_device *rpdev)
 		pm_heart_beat_work_handler);
 
 	pm_rpmsg.first_flag = true;
-	schedule_delayed_work(&heart_beat_work,
-			msecs_to_jiffies(100));
+	schedule_delayed_work(&heart_beat_work, 0);
 
 	pm_rpmsg.restart_handler.notifier_call = pm_restart_handler;
 	pm_rpmsg.restart_handler.priority = 128;
-- 
1.7.9.5

